msal/src/main/java/com/microsoft/identity/client/PublicClientApplication.java

```diff
@@ -231,15 +318,168 @@ public static IMultipleAccountPublicClientApplication createMultipleAccountPubli
      * @see <a href="https://developer.android.com/guide/topics/resources/providing-resources">Android app resource overview</a>
      * @see <a href="https://github.com/AzureAD/microsoft-authentication-library-for-android/wiki">MSAL Github Wiki</a>
      */
+    @WorkerThread
     public static IMultipleAccountPublicClientApplication createMultipleAccountPublicClientApplication(@NonNull final Context context,
-                                                                                                       final File configFile) {
+                                                                                                       @NonNull final File configFile) throws InterruptedException, MsalException {
+
+        PublicClientApplicationConfiguration configuration = loadConfiguration(configFile);
+        IPublicClientApplication application = createPublicClientApplication(context, configuration);
+        if(application instanceof IMultipleAccountPublicClientApplication){
+            return (IMultipleAccountPublicClientApplication)application;
+        }else{
+            if(configuration.mAccountMode == AccountMode.MULTIPLE && application.getIsSharedDevice())  {
+                throw new MsalClientException("AccountMode in configuration is set to multiple; however the device is marked as shared");
+            }
+            if(configuration.mAccountMode != AccountMode.MULTIPLE){
+                throw new MsalClientException("AccountMode in configuration is not set to multiple");
+            }
+            return null;
```

Comment:
Same question here - feels like it needs a fall-through error

Reply:
+1

Reply:
fixed.